libcfnet/stat_cache.c

```diff
@@ -147,9 +148,17 @@ int cf_remote_stat(AgentConnection *conn, bool encrypt, const char *file,
                  (intmax_t) tloc, file);
         int cipherlen = EncryptString(out, sizeof(out), in, strlen(in) + 1,
                                       conn->encryption_type, conn->session_key);
+
+        tosend = cipherlen + CF_PROTO_OFFSET;
+
+        if(tosend > sizeof(sendbuffer))
+        {
+            ProgrammingError("cf_remote_stat: tosend (%d) > sendbuffer (%ld)",
+                             tosend, sizeof(sendbuffer));
```

Comment:
FatalError() won't work as it takes EvalContext which is not available here. Prototype:

void FatalError(const EvalContext *ctx, char *s, ...) FUNC_ATTR_NORETURN FUNC_ATTR_PRINTF(2, 3);

Furthermore I don't see any benefit in returning here. Why would we do that? If it's reached it seems to be a bug and we should stop.